python/GafferSceneTest/RenderControllerTest.py

```diff
@@ -527,5 +527,217 @@ def testNullObjects( self ) :
+		controller.update()
+		assertMotionSamples( [ 1 ], False )
+		sphereAttributes['attributes']['transformBlur']["enabled"].setValue( True )
+		controller.update()
+		assertMotionSamples( [ 0.6, 1.4 ], False )
+
+		groupAttributes['attributes']['deformationBlur']["enabled"].setValue( True )
+		groupAttributes['attributes']['deformationBlur']["value"].setValue( False )
+		controller.update()
+		assertMotionSamples( [ 1 ], True )
+		sphereAttributes['attributes']['deformationBlur']["enabled"].setValue( True )
+		controller.update()
+		assertMotionSamples( [ 0.6, 1.4 ], True )
+
+		groupAttributes['attributes']['transformBlurSegments']["enabled"].setValue( True )
+		groupAttributes['attributes']['transformBlurSegments']["value"].setValue( 4 )
+		groupAttributes['attributes']['deformationBlurSegments']["enabled"].setValue( True )
+		groupAttributes['attributes']['deformationBlurSegments']["value"].setValue( 2 )
+		controller.update()
+		assertMotionSamples( [ 0.6, 0.8, 1.0, 1.2, 1.4 ], False )
+		assertMotionSamples( [ 0.6, 1.0, 1.4 ], True )
+
+		sphereAttributes['attributes']['transformBlurSegments']["enabled"].setValue( True )
+		sphereAttributes['attributes']['transformBlurSegments']["value"].setValue( 2 )
+		sphereAttributes['attributes']['deformationBlurSegments']["enabled"].setValue( True )
+		sphereAttributes['attributes']['deformationBlurSegments']["value"].setValue( 4 )
+		controller.update()
+		assertMotionSamples( [ 0.6, 1.0, 1.4 ], False )
+		assertMotionSamples( [ 0.6, 0.8, 1.0, 1.2, 1.4 ], True )
+
+		groupAttributes['attributes']['transformBlur']["value"].setValue( True )
+		groupAttributes['attributes']['deformationBlur']["value"].setValue( True )
+		sphereAttributes['attributes']['transformBlur']["value"].setValue( False )
+		sphereAttributes['attributes']['deformationBlur']["value"].setValue( False )
+		controller.update()
+		#assertMotionSamples( [ 0.6, 0.8, 1.0, 1.2, 1.4 ], False )
+		assertMotionSamples( [ 1.0 ], False )
+		assertMotionSamples( [ 1.0 ], True )
+
+		# Apply transformation to group instead of sphere, giving the same results
+		sphere["transform"]["translate"]["x"].setInput( None )
+		sphere["transform"]["translate"]["x"].setValue( 0 )
+		group["transform"]["translate"]["x"].setInput( frame["output"] )
+
+		groupAttributes['attributes']['transformBlur']["value"].setValue( True )
+		sphereAttributes['attributes']['transformBlur']["value"].setValue( False )
+		sphereAttributes['attributes']['transformBlurSegments']["enabled"].setValue( False )
+		controller.update()
+		assertMotionSamples( [ 0.6, 0.8, 1.0, 1.2, 1.4 ], False )
+
+		# Override transform segments on sphere
+		sphereAttributes['attributes']['transformBlur']["value"].setValue( True )
+		sphereAttributes['attributes']['transformBlurSegments']["enabled"].setValue( True )
+		sphereAttributes['attributes']['transformBlurSegments']["value"].setValue( 1 )
+		controller.update()
+		# Very counter-intuitively, this does nothing, because the sphere is not moving
```

Comment:
Fair enough that based on a hierachy based mental model this would be the expected result - what I don't like it that it can switch back and forth depending on whether animated samples of the child happen to be identical within this shutter. 
 Imagine a helicopter blade where the parent is spinning, and the child is animating tilt.  It would be quite confusing when on frame where the tilt changes, you lose all your motion blur samples - whereas if the child segments always overrode, it wouldn't match a hiearchy based mental model, but at least it would be predictable and easy to explain.

Reply:
That's not how the existing batch output works though is it? If a child has non-changing samples, we're keeping the samples from the parent, and concatenating the single child transform with each one.

Reply:
I'm describing the same thing as how batch currently works.  If the child has non-changing samples, we keep the parent samples, if it has changing samples, we overwrite them.  So depending on whether an animated object happens to move on a frame, we may get completely different sampling.

Reply:
Right, got you - I had completely misread what you were saying. Yep, that is definitely counterintuitive. Interesting that it hasn't come up in practice though. I think we're OK to keep current behaviour for this PR, and we're in a better position to change behaviour in future now RendererAlgo has been privatised.